### PR TITLE
Update the Chart to the latest PSONO release

### PR DESCRIPTION
**Pull request by PeeK1e**
Hi,

if this project is still alive, we would like to contribute back to the chart.

## What changed
* We created a secret scraper, that allows to generate the psono secrets the first time you install PSONO
  * a helper container is used for that [psono-helm-secret-helper](https://github.com/cloudical-io/psono-helm-secret-helper)
 ```yaml
 # -- Secret Creation Config
# Set options to Generate a Secret on first install
# It will only run if you install the chart for the first time!
# The following Secrets will be created
# PSONO_SECRET_KEY
# PSONO_ACTIVATION_LINK_SECRET
# PSONO_DB_SECRET
# PSONO_EMAIL_SECRET_SALT
# PSONO_PRIVATE_KEY
# PSONO_PUBLIC_KEY
secrets:
  # Set to false if you have your own secrets.
  # If you have predefined values, you need to mount them in the sections below
  # or define them in the env section
  create: true
  # Set the Name of the generated secrets
  name: "psono-deploy-secrets"
  ```
* Secrets for the postgres can be referenced from a secret already in the cluster
```yaml
# -- Database credentials from Secret
databaseSecret:
  # -- enables or disables the secret to be used
  enabled: false
  # -- secret name
  name: "psono-postgres-secret"
  # -- Username Key Name
  usernameKey: "username"
  # -- Password Key Name
  passwordKey: "password"
  ```
  * The Ingress was updated to handle the new routing style of PSONOs' Server
    * It takes the `PSONO_URL_PREFIX` variable that was recently introduced into PSONO

### CODE REVIEW

**Comment by tiagoposse** [comment]
Hi!
This project is indeed alive. I'm just running an older version, so haven't felt the need to update this yet.
Thank you for contributing!
Have a few requests:
- would you mind targeting the PR against "dev"? There are a few more changes I want to do before releasing a new version.
- would you be okay with having the code present in https://github.com/cloudical-io/psono-helm-secret-helper as a configmap in the chart, using an image like python:xx in the secret creation job. This would make sure the chart is self-contained.

Will leave more comments throughout

File: templates/server.yaml [hunk 1]
```diff
@@ -32,8 +32,11 @@ spec:
             - sh
             - -c 
             - >
+              {{- if .Values.server.envFiles }}
               export $(xargs < {{ join " " .Values.server.envFiles }});
-              /root/configs/docker/cmd.sh
+              {{- end }}
+              env;
```
Comment (tiagoposse) [inline templates/server.yaml:38]: this will expose the secrets

File: Chart.yaml [hunk 2]
```diff
@@ -15,15 +15,15 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.2.0
+version: 0.2.3
```
Comment (tiagoposse) [inline Chart.yaml:18]: seems to not be consistent with the changelog. Could all changes in this MR be put together into 0.2.1?

Comment (PeeK1e) [reply Chart.yaml:18]: Need to look into this there might be a change that I forgot to add into the Changelog

File: templates/server.yaml (hunk 1 above)
Comment (PeeK1e) [reply templates/server.yaml:38]: Might have forgotten the `env;` from debugging the deployment

**Comment by PeeK1e** [comment]
> Hi!
> This project is indeed alive. I'm just running an older version, so haven't felt the need to update this yet.
> Thank you for contributing!
> Have a few requests:
> - would you mind targeting the PR against "dev"? There are a few more changes I want to do before releasing a new version.
> - would you be okay with having the code present in https://github.com/cloudical-io/psono-helm-secret-helper as a configmap in the chart, using an image like python:xx in the secret creation job. This would make sure the chart is self-contained.
> 
> Will leave more comments throughout

Yeah I can change the target branch later (probably Monday).

Regarding the image, I would have to check if we could use a python image because we need access to the API via kubectl. I'm sure you could use API calls or a python lib to make the secret creation calls. But for now this should suffice :)

**Pull request closed without merging**